jpo-ode-svcs/src/test/java/us/dot/its/jpo/ode/coder/stream/LogFileToAsn1CodecPublisherTest.java

```diff
@@ -15,50 +15,332 @@
 import us.dot.its.jpo.ode.importer.parser.FileParser.ParserStatus;
 import us.dot.its.jpo.ode.importer.parser.LogFileParser;
+import us.dot.its.jpo.ode.model.OdeData;
+import us.dot.its.jpo.ode.model.OdeLogMetadata.RecordType;
+import us.dot.its.jpo.ode.model.OdeMsgMetadata;
+import us.dot.its.jpo.ode.util.DateTimeUtils;
 
 public class LogFileToAsn1CodecPublisherTest {
 
-   @Tested
+   private static final String GZ = ".gz";
+
+  @Tested
    LogFileToAsn1CodecPublisher testLogFileToAsn1CodecPublisher;
 
    @Injectable
    StringPublisher injectableStringPublisher;
+   
+   @BeforeClass
+   public static void setupClass() {
+     OdeMsgMetadata.setStaticSchemaVersion(OdeProperties.OUTPUT_SCHEMA_VERSION);
+   }
+   
+   @Test
+   public void testPublishInit(@Mocked LogFileParser mockLogFileParser) throws Exception {
+      new Expectations() {
+         {
+            LogFileParser.factory(anyString);
+            result = mockLogFileParser;
 
-   @Capturing
-   LogFileParser capturingLogFileParser;
+            mockLogFileParser.parseFile((BufferedInputStream) any, anyString);
+            result = ParserStatus.INIT;
+         }
+      };
 
-   @Mocked
-   LogFileParser mockTimLogFileParser;
+      List<OdeData> dataList = testLogFileToAsn1CodecPublisher.publish(new BufferedInputStream(new ByteArrayInputStream(new byte[0])),
+            "fileName", ImporterFileType.LEAR_LOG_FILE);
 
+      assertTrue(dataList.isEmpty());
+   }
+   
    @Test
-   public void testPublishEOF() throws Exception {
+   public void testPublishEOF(@Mocked LogFileParser mockLogFileParser) throws Exception {
       new Expectations() {
          {
             LogFileParser.factory(anyString);
-            result = mockTimLogFileParser;
+            result = mockLogFileParser;
 
-            mockTimLogFileParser.parseFile((BufferedInputStream) any, anyString);
+            mockLogFileParser.parseFile((BufferedInputStream) any, anyString);
             result = ParserStatus.EOF;
          }
       };
 
+      List<OdeData> dataList = testLogFileToAsn1CodecPublisher.publish(new BufferedInputStream(new ByteArrayInputStream(new byte[0])),
+            "fileName", ImporterFileType.LEAR_LOG_FILE);
+
+      assertTrue(dataList.isEmpty());
+   }
+   
+   @Test(expected = Exception.class)
+   public void testPublishThrowsException(@Mocked LogFileParser mockLogFileParser) throws Exception {
+      new Expectations() {
+         {
+            LogFileParser.factory(anyString);
+            result = mockLogFileParser;
+
+            mockLogFileParser.parseFile((BufferedInputStream) any, anyString);
+            result = new Exception();
+         }
+      };
+
       testLogFileToAsn1CodecPublisher.publish(new BufferedInputStream(new ByteArrayInputStream(new byte[0])),
-            "fileName", ImporterFileType.OBU_LOG_FILE);
+            "fileName", ImporterFileType.LEAR_LOG_FILE);
+      fail("Expected an Exception to be thrown");
```

Comment:
Why is publish supposed to throw an exception here? What type of exception? Might be good to specify a little more detail about why an exception is being thrown here, and specifying the type of the exception will hint at what we expect to go wrong. Many things can cause an exception, so it is helpful to identify exactly what.

Reply:
Will add comments and a new test case. Hopefully answers your question.